### DISCUSSION TITLE
How update 1.4 or 2.0 to Evolution CMS 3.0

### DISCUSSION THREAD

**Discussion by nick0** [evolution-cms/evolution#1897] [Q&A]
Hello,

Why is Evo at 3 now? What was wrong with 2?

Before you develop further, can we please have an update tool to update 1.4.12 to 2.x from the Manager home page like we can with sub versions when updating 1.4.x? And also then one for 2.x to 3.x. Without that, this feels like Evo to Revo all over again, only worse because now there is an extra version to update through,

Cheers.

**Comment by Dmi3yy** [comment]
Hi, today will video how update 1.4 and 2.0 to 3.0 

Add subtitles in English soon. 

Not see any problem with update. But need run migration script.

Soon will finished new version of FormLister after that, 3.0 have full compatibility with 1.4 and 2.0 ( i think it will on this week )

**Comment by ArnoNymeier** [comment]
Hello,
i think video is not a good way to understand things. I tried to follow some of the already existing videos, no chance. everything too fast and not understandable.
i doubt the subtitles make it more convenient.

What speaks against a file with step-by-step instructions that can be processed comfortably and always at a glance?

**Comment by Dmi3yy** [comment]
Plz watch this video: 
https://www.youtube.com/watch?v=RjOiq35FgW4

I add some English text on Video, and add english subtitles, i think its help for better understand how all this work. 

If have comments how do video more comfortable plz write me this. 

I think the video conveys more information. Anyone can translate the video into an article, unfortunately I don't have enough time to do everything myself. I also want to remind you that this is open source and anyone can help with development.

Now there is a great need for translation of documentation and lessons into English. And better if this do someone who good know english ).

**Comment by fourroses666** [comment]
@nick0 
Upgrading for non-tech users can be pretty easily done with the installer file.
https://github.com/evolution-cms/installer

The steps I do when updating from 1 -> 2 is: 
- backup
- use installer (install.php in root) and navigate to www.site.com/install.php
- because the config file is moved you need to add this file in you folders. This is the only "difficult step"
What I do is run the installer 2x and then add the config file. 
But you can also add the file before upgrading yourself and run installer just once..
Without that file the installer can do only a fresh install, doesn't know there is an already installed version.
So you need to add the folders and config file: /core/config/database/connections/default.php
The default.php looks like this:
```
<?php
return [
    'driver' => env('DB_TYPE', 'mysql'), //$database_type
    'host' => env('DB_HOST', 'localhost'), //$database_server
    'port' => env('DB_PORT', '3306'),
    'database' => env('DB_DATABASE', 'dbnameeee'), //$dbase
    'username' => env('DB_USERNAME', 'dbuserrrr'), //$database_user
    'password' => env('DB_PASSWORD', 'passssss'), //$database_password
    'unix_socket' => env('DB_SOCKET', ''),
    'charset' => env('DB_CHARSET', 'utf8mb4'), // $database_connection_charset
    'collation' => env('DB_COLLATION', 'utf8mb4_general_ci'), //$database_collation
    'prefix' => env('DB_PREFIX', 'evo_'), //$table_prefix
    'method' => env('DB_METHOD', 'SET NAMES'), //$database_connection_method
    'strict' => env('DB_STRICT', false),
    'engine' => env('DB_ENGINE'),
    'options' => [
        PDO::ATTR_STRINGIFY_FETCHES => true,
    ]
];

```
You can copy the dbname+pass+prefix from the old config file /manager/includes/config.inc.php
ps. When upgrading within EVO 2 the default.php file needs CHMOD 644 to work. Looks like its set to 404 and can't be read.
- Run the installer
- When done I always clear /assets/cache/ (siteCache.idx.php+siteManager.php)
- You might need to update some Extras, just fully check your site.

Not sure if .htaccess is overwritten but I don't think so.


Updating from 2 -> 3 can also be done with the installer.
You should wait till 3.0.1 is there (very soon) because it has an important fix (snippet caching or something like that).

**Comment by Dmi3yy** [comment]
Upgrade from 1.4 and 2.0 to 3.0 only by Migration script or will not work with users :) see video: 
https://www.youtube.com/watch?v=RjOiq35FgW4 
with subtitles )

**Comment by fourroses666** [comment]
@nick0 
Upgrading 2 -> 3 isn't hard too. I should wait till 3.0.1 is there because of a little bugfix when using snippets in snippets, (something I had issues with in my site, IF in PageBuilder and is fixed in 3.0.1).
- backup
- download "Migrate 1.4.12 and 2.0.4 to 3.0" from Extras
- select Default theme on Configuration
- clear cache and log out
- run/go to site.com/assets/update.php
- install 3.0
- might be you need to CHMOD the config file: /core/config/database/connections/default.php to 644 if its 404
- I think thats all. 
- you can delete of files in the root, see last seconds of the Youtube video :) https://www.youtube.com/watch?v=RjOiq35FgW4

All snippets and plugins are removed from core so you existing plugins/snippets might need to be updated in Extras

p.s.
The WEBuser system is changed in EVO 3 so if you have somekind of user login website there needs to be done some changes. Use FormLister I think but I have no idea myself because I didn't do it myself yet.

**Comment by fourroses666** [comment]
ps. There was nothing wrong with EVO 2 but the core is changed allot. Laravel is fully integrated and if you watch the video you know more whats going on. https://www.youtube.com/watch?v=xDIdfZrlCgk 

Maybe its better to change the title in this post to. How to update from EVO 1 to EVO 3

**Comment by nick0** [comment]
Hi @fourroses666 - thank you so much for the step by step instructions,
It looks a little daunting but will give it a go (likely next week) per your tips. Cheers mate.

> What speaks against a file with step-by-step instructions that can be processed comfortably and always at a glance?

100% agree with @ArnoNymeier . Especially when its quite involved, step by step (like Marc's notes) is much easier to work through. Even better would be a little install package that works from the Manager home page.

> Maybe its better to change the title...

To me the title shouldn't change. If 2 was the new Evo, why is it now 3.0.1 and not 2.0,4? 3.0.x with important snippets apparently not working to me seems weird - I would have thought fast / forward development without backward compatability of the fundamental snippets seems unusual. That's what I don't understand - why was the 2.0.x version so shortlived?

> Now there is a great need for translation of documentation and lessons into English. And better if this do someone who good know english ).

I understand what you are saying and that docs are not your focus but surely it needs to be done by the developers... you are the ones that know everything about Evo and the snippets and all the changes etc.

**Comment by fourroses666** [comment]
We can place the Upgrade method/steps in the new Forum beside the video on youtube.
https://forum.evo.im/

If there are any issues it can be posted there (or here on Github if its a bug)

Can't tell why but I think it just had to be done allot different and needed a new version because 2.0 was already there.

Google translation will help allot with translation. If I have a little more time I can help translating, at the moment too much going on. Its open source so we need to help all a little. I'm very thankful to Dmytro and Co to keep this alive after being abandoned by modx. This without sponsorship and many hours of work.

**Comment by nick0** [comment]
Thanks @fourroses666. 

That is an excellent idea on adding the upgrade method / steps into the new forum :)

Ah right. I am interested to hear the reasoning of the short lifespan of 2.x and then to 3.x from @Dmi3yy.
I am not a programmer so don't understand the need for Laveral - I am all about the frontend templating, DocManager and the primary snippets.

Agreed. Very grateful to all the devs and contributors to keep Modx Evolution > Evolution CMS alive.

**Comment by fourroses666** [comment]
I think this can be closed, I have added it to the forum.

https://forum.evo.im/d/5-how-to-upgrade-evo-1-to-evo-2
https://forum.evo.im/d/49-how-to-upgrade-evo-2-to-evo-3

**Comment by kamaworld** [comment]
I'm Japanese.
I'm not fluent in English so it's hard to understand on YouTube.
If this explanation is in text, you can learn it with Google Translate.
Please explain in text so that you can understand it in various languages.
I think you should upload this method to the Evolution CMS site so that anyone can understand it.

I used to use MODX a long time ago. I decided to use it again.
However, the MODX forums and updates in Japan have stopped.
I decided to build it with an Evolution CMS.
I'm not a good engineer, but I understand the wonders of Evolution CMS.
I think that the number of Japanese people who use Evolution CMS will increase in the future.
Thank you.
